### PR TITLE
changelog update v0.4.9

### PR DESCRIPTION
**Pull request by geoffthemedio**
changelog update v0.4.9

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -3,6 +3,198 @@
...
+-Reduced AI invasion overkill / waste
+-Made AI consider lost fleets when determining unknown threats
+
+#### Bugs
+
+-Fixed potential crash in password entry box
+-Fixed log level filters briefly not applying while being set
+-Fixed laggy UI updates when resizing the sitreps window
+-Fixed potential crash when setting fleet routes
+-Fixed issue where genreated filename timestamps could include invalid filename characters on systems with some languages
```
Comment (geoffthemedio) [inline ChangeLog.md]: generated

File: ChangeLog.md [hunk 2]
```diff
@@ -3,6 +3,198 @@
...
+-Fixed issue where empire colours in a saved game could conflict with player colours in the multiplayer lobby
+-Fixed server crash when receiving turn orders from incompatible client
+-Fixed issue where shield regeneration could apply twice
+-Fixed issue where monster shields wouldn't regenerate
+-Fixed crash when editing long strings in a narrow window
+-Fixed issue where certain ship part meter effects could not work
+-Fixed issue where scrap orders couldn't be issued
+-Made unowned fleets move before empire fleets, which avoids some issues with blockades
+-Fixed issue with blockade determination so that blockaded planets are still connected to themselves
+-Fixed issue with fleets that had never moved being unable to enforce bloackades
```
Comment (geoffthemedio) [inline ChangeLog.md]: blockades

File: ChangeLog.md [hunk 3]
```diff
@@ -3,6 +3,198 @@
...
+-Fixed issue where loaded games weren't setting the server rules to those in the save
+-Resolved some issues with premade ship designs being added (or not added) at start of game
+-Fixed potential issues with invalid UTF8 strings when doing multi-line text layout
+-Fixed issue with incorrect refueling calculations when determining move pathes for fleets
```
Comment (geoffthemedio) [inline ChangeLog.md]: paths

File: ChangeLog.md [hunk 4]
```diff
@@ -3,6 +3,198 @@
...
+-Added an option to set a game unique ID number, which is useful for multiplayer servers
+-Python errors are output to logging instead of stderr
+-Hostless servers autosave when shutting down
+-Added an OrderedAlternativesOf condition that will use the first condition in its list of subconditions that matches something
+-Added SVG rendering in pedia, and allowed SVG images to be included in articles
+-Reworked timing of clients sending save state info to avoid potential save corruption
+-A warning popup should be shown if the client's OpenGL version is < 2.0
+-Added mechanism for chat messages to have automatic string formatting and stringtable substitutions when displayed
+-Added an option to enable/disable network binary serialization
+-Screensaver is no longer enabled/disabed in response to window minimize, maximize, or restore events
```
Comment (geoffthemedio) [inline ChangeLog.md]: disabled

**Comment by o01eg** [comment]
Not sure if it's a key change, but there was introduced "in-game" multiplayer lobby where non-game player can talk in the chat or choose empire to play if it allowed.

**Comment by geoffthemedio** [comment]
> "in-game" multiplayer lobby

I wasn't really sure how to describe that when adapting the commit logs to change logs. Seems like it's part of having a persistent server that one or more players can join and play their turns... First you go to the lobby, then pick an empire and play the turn. Doesn't matter if someone else is already connected, I suppose?

**Comment by o01eg** [comment]
> Doesn't matter if someone else is already connected, I suppose?

No, someone else could join to lobby and use chat.

Also, it can be used in a public server when someone can ask players to start new game with him.

**Comment by geoffthemedio** [comment]
It currently says "Chat history ... is shared between the lobby and in-game" which sort of covers that? Or do you want to elaborate on that...?

**Comment by o01eg** [comment]
It suppose it related to copying chat content from lobby to game chat, not "in-game" lobby concept itself.

File: ChangeLog.md [hunk 5]
```diff
@@ -3,6 +3,198 @@
...
 
+## [v0.4.9] - 2019-12-??
+
+#### Key Changes
+
+- Multiplayer enhancements
+-- Hostless servers can be started without connected players and left running for players to join to play their turns asynchronously
+-- Empires in a game can be restricted to certain players based on username-password authentication
+-- Known / previously joined servers are saved and appear in the servers list of the connection window
+-- Chat history persists and is sent to players that connect to a server, and is shared between the lobby and in-game
```
Comment (geoffthemedio) [inline ChangeLog.md]: Maybe break up these points?
````
--Chat is shared between the lobby and in-game
--Chat history persists and is sent to players that connect to a server
````

File: ChangeLog.md [hunk 6]
```diff
@@ -3,6 +3,198 @@
...
+-- The chat / message window flashes when a message is received
+-- The lobby shows galaxy setup data from a loaded save
+-- The lobby shows empires without assigned players as separate rows
```
Comment (geoffthemedio) [inline ChangeLog.md]: Maybe elaborate / rephrase these?

````
-- The lobby shows galaxy setup data for a loaded save or the game being played on the server
-- The lobby shows empires without assigned players as separate rows when loading a save or for the game being played on the server
````

File: ChangeLog.md (hunk 5 above)
Comment (o01eg) [reply ChangeLog.md]: Yes, I think it is better.

File: ChangeLog.md (hunk 6 above)
Comment (o01eg) [reply ChangeLog.md]: Yep.

**Comment by geoffthemedio** [comment]
tweaked contents and list formatting

**Comment by Vezzra** [comment]
@freeorion/contributors, @freeorion/developers: Any comments on the proposed changelog? Unless anyone raises any objections or requests corrections, I'm going to merge this coming Sunday.

**Pull request merged**